yb-voyager/src/srcdb/postgres.go

```diff
@@ -1735,3 +1824,238 @@ func (pg *PostgreSQL) GetSchemasMissingUsagePermissions() ([]string, error) {
+	if endpointsStr == "" {
+		return nil, nil
+	}
+
+	endpoints := strings.Split(endpointsStr, ",")
+	var replicaEndpoints []ReplicaEndpoint
+
+	for i, endpoint := range endpoints {
+		endpoint = strings.TrimSpace(endpoint)
+		if endpoint == "" {
+			continue
+		}
+
+		var host string
+		var port int = 5432 // default port
+
+		// Simple host:port parsing (consistent with primary connection)
+		parts := strings.Split(endpoint, ":")
+		if len(parts) == 1 {
+			host = parts[0]
+		} else if len(parts) == 2 {
+			host = parts[0]
+			var err error
+			port, err = strconv.Atoi(parts[1])
+			if err != nil || port <= 0 || port > 65535 {
+				return nil, fmt.Errorf("invalid port in endpoint at position %d: %s", i+1, parts[1])
+			}
+		} else {
+			return nil, fmt.Errorf("malformed endpoint at position %d: %s (expected format: host[:port])", i+1, endpoint)
+		}
+
+		if host == "" {
+			return nil, fmt.Errorf("empty host in endpoint at position %d", i+1)
+		}
+
+		// Use endpoint string as default name (will be enriched with application_name if available)
+		name := fmt.Sprintf("%s:%d", host, port)
+		replicaEndpoints = append(replicaEndpoints, ReplicaEndpoint{
+			Host:          host,
+			Port:          port,
+			Name:          name,
+			ConnectionUri: pg.GetReplicaConnectionUri(host, port),
+		})
+	}
+
+	// De-duplicate endpoints
+	seen := make(map[string]bool)
+	var uniqueEndpoints []ReplicaEndpoint
+	for _, ep := range replicaEndpoints {
+		key := fmt.Sprintf("%s:%d", ep.Host, ep.Port)
+		if !seen[key] {
+			seen[key] = true
+			uniqueEndpoints = append(uniqueEndpoints, ep)
+		}
+	}
+
+	return uniqueEndpoints, nil
+}
+
+// EnrichReplicaEndpointsWithDiscovery matches provided endpoints with discovered replicas
+// and enriches the Name field with application_name when available.
+// Note: We cannot assume discovered replicas and provided endpoints are in the same sequence.
+// Matching strategy: Try exact match by host address (client_addr from discovery vs host from endpoint).
+// If a match is found and application_name is available, use it; otherwise keep endpoint string (host:port).
+func EnrichReplicaEndpointsWithDiscovery(endpoints []ReplicaEndpoint, discoveredReplicas []ReplicaInfo) []ReplicaEndpoint {
+	enriched := make([]ReplicaEndpoint, len(endpoints))
+	copy(enriched, endpoints)
+
+	// Create a map of discovered replicas by client_addr for quick lookup
+	// Key: client_addr, Value: ReplicaInfo
+	replicaByAddr := make(map[string]ReplicaInfo)
+	for _, replica := range discoveredReplicas {
+		if replica.ClientAddr != "" {
+			replicaByAddr[replica.ClientAddr] = replica
+		}
+	}
+
+	// Try to match each endpoint with discovered replicas by exact address match
+	for i := range enriched {
+		ep := enriched[i]
+
+		// Try exact match by host address
+		if replica, found := replicaByAddr[ep.Host]; found {
+			if replica.ApplicationName != "" {
+				enriched[i].Name = replica.ApplicationName
```

Comment:
Why do we need to populate this populating Name field from discovered replicas? Where do we use the application name field?